integrationtest/src/it/scala/io/gearpump/integrationtest/checklist/RestServiceSpec.scala

```diff
@@ -142,18 +143,64 @@ class RestServiceSpec extends TestSpecBase {
 
   "cluster information" should {
     "retrieve 1 master for a non-HA cluster" in {
+      // setup
+      val expectMastersCount = cluster.getMasterHosts.size
+      val expectMaster = cluster.getMasters.head
 
-    }
-
-    "retrieve 0 worker, if cluster is started without any workers" in {
-      // todo: defect
+      // exercise
+      restClient.listMasters().size shouldEqual expectMastersCount
+      restClient.listMasters().head shouldEqual expectMaster
+      restClient.queryMaster().aliveFor should be > 0L
+      restClient.queryMaster().masterStatus shouldEqual MasterStatus.Synced
     }
 
     "retrieve the same number of workers as cluster has" in {
+      // setup
+      val expectWorkersCount = cluster.getWorkerHosts.size
+
+      // exercise
+      restClient.listRunningWorkers().size shouldEqual expectWorkersCount
+      restClient.listRunningWorkers().foreach { worker =>
+        worker.state shouldEqual MasterToAppMaster.AppMasterActive
+      }
     }
 
     "find a newly added worker instance" in {
+      // setup
+      val oldWorkersCount = restClient.listRunningWorkers().size
+      val workerName = s"newWorker"
```

Comment:
The modifier `s` is not necessary. `s"newWorker"` could be `"newWorker". Please check IDE warnings before you commit code. We have a great IntelliJ to find warnings for you. 

Not an obvious issue but how can you be sure, there is no existing worker called "newWorker"?